mlflow/entities/experiment.py

```diff
@@ -7,14 +8,13 @@ class Experiment(_MLflowObject):
     Experiment object.
     """
     DEFAULT_EXPERIMENT_ID = 0
-    ACTIVE_LIFECYCLE = 'active'
-    DELETED_LIFECYCLE = 'deleted'
 
     def __init__(self, experiment_id, name, artifact_location, lifecycle_stage):
         super(Experiment, self).__init__()
         self._experiment_id = experiment_id
         self._name = name
         self._artifact_location = artifact_location
+        assert(LifecycleStage.is_valid(lifecycle_stage))
```

Comment:
This seems mostly safe AFAICT (correct me if I'm wrong) - when loading experiments from the FileStore, we fill in `lifecycle_stage` based on the directory the experiment is in ([see code](https://github.com/mlflow/mlflow/blob/master/mlflow/store/file_store.py#L196)), so this assertion should pass. 

The one place I could see this breaking however is when deserializing experiments from an old server that doesn't return a lifecycle stage (for example, when calling `MlflowClient().list_experiments` or `MlflowClient()get_experiment()` against an old tracking server). However we've had the notion of experiment lifecycle stages (support for deleted runs) since 0.6.0 in mid-September, so maybe this breaking change is ok - probably worth noting in the changelog though.

Note that the same applies for this assertion in RunInfo (we've had lifecycle stages for runs since 0.7.0 in early October), which would break calling `mlflow.log_artifact`, `mlflow.get_run` etc against an old tracking server ([link](https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/client.py#L161))

Actually thinking about it maybe we should just defer asserting here till 1.0, since I don't think reading/persisting a bad lifecycle stage is a major concern given the current APIs

Reply:
Good catch -- fixed